sdb/commands/internal/table.py

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+from typing import Any, Callable, Dict, List, Optional, Set
+
+
+class Table:
+    """
+    Generic Table implementation for pretty-printing
+    data in columns.
+    """
+
+    __slots__ = "fields", "rjustfields", "formatters", "maxfieldlen", "lines"
+
+    def __init__(self,
+                 fields: List[str],
+                 rjustfields: Optional[Set[str]] = None,
+                 formatters: Optional[Dict[str, Callable[[Any], str]]] = None
+                ) -> None:
+        self.fields = fields
+
+        if rjustfields is None:
+            self.rjustfields = ()
+        else:
+            self.rjustfields = rjustfields
+
+        self.formatters = {field: str for field in fields}
+        if not formatters is None:
+            for field in formatters.keys():
+                if field in self.fields:
+                    self.formatters[field] = formatters[field]
```

Comment:
I wonder if it would be easier to understand this as:

```
self.formatters = {}
for field in fields:
  if field in formatters.keys():
    self.formatters[field] = formatters[field]
  else
    self.formatters[field] = str
```

Or maybe I'm just not thinking "pythonic" enough.  There's probably a dictionary comprehension that's even more compact.